### ISSUE TITLE
Status update?

### ISSUE THREAD

**Issue by lukehutch** [NitorCreations/vertx-acme4j#1]
This project is marked deprecated -- is that because it is no longer maintained, or because SNI-01 validation was shut down, or because there's a new, better way to do ACME validation for Vert.x?

**Comment by xkr47** [comment]
Hi,

Thanks for your interest. It was marked deprecated when SNI-01 was shut down. I have not had time to reimplement it to use SNI-02 although that would probably be great. So my home servers are without https at the moment :/ I will however update the README to reflect this now.

I do not foresee myself having time to implement SNI-02 in 2018, but you never know.

The [AcmeManager](https://github.com/xkr47/vertx-acme4j/blob/master/src/main/java/space/xkr47/vertx/acme4j/AcmeManager.java) code is ugly because it was at the same time an academic effort to write everything in async :-) Java does not bend..

**Comment by lukehutch** [comment]
Thanks for the update. Exactly because of "nested callback hell" I have been looking for ways to use Vert.x that do not involve callbacks, and I came up with the combination of the following two projects -- an async/await implementation for Java based on CompletableFutures / promises, and a CompletableFuture API for Vert.x. I have yet to test if this combination works, but it looks promising, no pun intended:

* https://github.com/electronicarts/ea-async
* https://github.com/cescoffier/vertx-completable-future

This may make the SNI-02 rewrite a lot more palatable.

**Comment by xkr47** [comment]
Thanks for the tips!

**Comment by lukehutch** [comment]
Actually this might be even better...

https://vertx.io/docs/vertx-sync/java/

**Comment by lukehutch** [comment]
@xkr47 I started looking into this again, and I thought of a simpler way to add ACME support to a Vertx project. Would it be possible to simply set up certbot as a cron job, and then use a cutdown version of vertx-acme4j that only consists of something like `DynamicCertManager` to watch for certificate changes? I don't know much about certificate management, but it seems like this should work.

(Instead, if you do still want to implement the full SNI-02 without all the callbacks, you could also try writing it in Kotlin, with [vertx-kotlin-coroutines](https://vertx.io/docs/vertx-lang-kotlin-coroutines/kotlin/).)

**Comment by xkr47** [comment]
Yeah you totally can use just the DynamicCertManager to (re)import certificates on-the-fly. It was even part of the plan. For people wanting to do non-SNI handshakes it is the way to go. For SNI it made sense (to me) to build a "all-in-one" solution. Also, it was a bit of a academic approach :)

I'll try to add some docs on how to use DynamicCertManager standalone. Also, feel free to request anything more you need. :)

**Comment by xkr47** [comment]
Ok done, lmk if you need code examples.. You can look in [AcmeManager.java](https://github.com/xkr47/vertx-acme4j/blob/master/src/main/java/space/xkr47/vertx/acme4j/AcmeManager.java) and search for `dynamicCertManager.`

**Comment by lukehutch** [comment]
Thanks. I think you should seriously consider upstreaming your dynamic certificate reloading code into Vertx:

https://github.com/eclipse-vertx/vert.x/issues/2870
